src/ngraph_encapsulate_op.cc

```diff
@@ -460,78 +460,19 @@ class NGraphEncapsulateOp : public OpKernel {
     return Status::OK();
   }
 
-  //---------------------------------------------------------------------------
-  // OpKernel::Compute
-  //---------------------------------------------------------------------------
-  void Compute(OpKernelContext* ctx) override {
-    std::ostringstream oss;
-    oss << "Execute: Encapsulate_" << my_instance_id << ": " << name();
-    ngraph::Event event(oss.str(), name(), "");
-
-    Timer compute_time;
-    std::lock_guard<std::mutex> lock(m_compute_lock);
-    NGRAPH_VLOG(4) << "NGraphEncapsulateOp::Compute starting for cluster "
-                   << m_ngraph_cluster;
-
-    ngraph::Event event_func_maybe_create("FunctionMaybeCreate", name(), "");
-    Timer function_lookup_or_create;
-
-    std::vector<TensorShape> input_shapes;
-    std::vector<const Tensor*> static_input_map;
-    std::shared_ptr<ngraph::Function> ng_function;
-    std::shared_ptr<ngraph::runtime::Executable> ng_exec;
-    ng::runtime::Backend* op_backend;
-
-    // Get ngraph executable and inputs information
-    OP_REQUIRES_OK(ctx, GetNgExec(ctx, ng_exec, input_shapes, static_input_map,
-                                  op_backend));
-
-    NGRAPH_VLOG(4)
-        << "NGraphEncapsulateOp::Compute got ngraph executable for cluster "
-        << m_ngraph_cluster;
-
-    int time_func_create_or_lookup = function_lookup_or_create.ElapsedInMS();
-    event_func_maybe_create.Stop();
-
-    NGRAPH_VLOG(4) << "NGraphEncapsulateOp::Compute got graph for cluster "
-                   << m_ngraph_cluster;
-
-    Timer create_or_lookup_tensors;
-
-    if (m_freshness_tracker == nullptr) {
-      auto creator = [](NGraphFreshnessTracker** tracker) {
-        *tracker = new NGraphFreshnessTracker();
-        return Status::OK();
-      };
-      OP_REQUIRES_OK(
-          ctx, ctx->resource_manager()->LookupOrCreate<NGraphFreshnessTracker>(
-                   ctx->resource_manager()->default_container(),
-                   "ngraph_freshness_tracker", &m_freshness_tracker, creator));
-    }
-
-    NGRAPH_VLOG(4)
-        << "NGraphEncapsulateOp::Compute got freshness tracker for cluster "
-        << m_ngraph_cluster;
-
+  Status AllocateTensorInput(
+      OpKernelContext* ctx,
+      std::shared_ptr<ngraph::runtime::Executable>& ng_exec,
+      std::vector<TensorShape>& input_shapes, ng::runtime::Backend* op_backend,
+      vector<shared_ptr<ng::runtime::Tensor>>& ng_inputs,
+      std::vector<std::unique_ptr<ngraph::Event>>& input_copy_events) {
```

Comment:
I think that `ng_exec`, `input_shapes`, `op_backend` and `ng_inputs` are the essential inputs to this function.
`input_copy_events` just happens to be the tracing instrumentation. in that case we should just time `AllocateTensorInput` outside at its call site.

```
start_timer(input_copy_events)
AllocateTensorInput()
stop_timer(input_copy_events)
```

instead of:
```
AllocateTensorInput(input_copy_events)
```

Same comment for `AllocateTensorOutput`

Comments @avijit-nervana ?

Reply:
I removed input_copy_events as input to this function and moved the for loop inside the function as @shresthamalik suggested instead of doing it in Compute.